template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/zeppelin/ZeppelinS3CloudStorageRoleConfigProvider.java

```diff
@@ -64,12 +64,13 @@ protected String getCloudStorageProperty(TemplatePreparationObject source) {
 
         ConfigUtils.getStorageLocationForServiceProperty(source, ZEPPELIN_NOTEBOOK_S3_BUCKET).ifPresent(
                 storageLocation -> hmsCloudStorage.append(
-                        ConfigUtils.getSafetyValveProperty(ZEPPELIN_NOTEBOOK_S3_BUCKET, storageLocation.getValue()))
+                        ConfigUtils.getSafetyValveProperty(ZEPPELIN_NOTEBOOK_S3_BUCKET,
+                                storageLocation.getValue().replace(S3A_PREFIX, "")))
         );
 
         String zeppelinUser = ConfigUtils.getStorageLocationForServiceProperty(source, ZEPPELIN_NOTEBOOK_S3_USER)
                 .map(StorageLocationView::getValue)
-                .orElse(ZEPPELIN_DEFAULT_USER);
+                .orElse(source.getGeneralClusterConfigs().getClusterName());
```

Comment:
Won't Zeppelin also hang if storage location for `zeppelin.notebook.s3.user` is present and the same value is given for more than one cluster?  In other words, do we need to add the cluster name only in fallback case?

Reply:
Yes you are correct. Updated zeppelin.notebook.s3.user  to always use `<clustername>/zeppelin`. So that the complete zeppelin notebook path becomes `<bucket>/<clustername>/zeppelin/notebook`